Codec/Compression/Zlib/Internal.hs

```diff
@@ -586,13 +588,38 @@ decompressST :: Stream.Format -> DecompressParams -> DecompressStream (ST s)
 --
 decompressIO :: Stream.Format -> DecompressParams -> DecompressStream IO
 
-decompress   format params = foldDecompressStreamWithInput
-                               L.Chunk (const L.Empty) throw
-                               (decompressStreamST format params)
+decompress   format params = unsafeFromChunks . safeDecompress format params
 decompressST format params = decompressStreamST  format params
 decompressIO format params = decompressStreamIO  format params
 
 
+-- | Like 'decompress' except that it returns the 'DecompressError' instead of
+-- throwing it.
+safeDecompress
+  :: Stream.Format
+  -> DecompressParams
+  -> L.ByteString
+  -> Chunks L.ByteString DecompressError
+safeDecompress format params = foldDecompressStreamWithInput
+  (Next . L.fromStrict)
```

Comment:
why `L.fromStrict` ? 

`unsafeFromChunks` using `<>` would have terrible performance (`n ^ 2`) in chunks

Also, I suspect that `decompress` allocates `Chunks` now, where old `decompress` folded into `Lazy.ByteString` without tweaking.

---

- I'd keep `decompress` implementation as it is. Then we can be 99.99% sure that it doesn't change.
- `safeDecompress :: ... -> Chunks S.ByteString DecompressError`, i.e. use strict bytestring as a single chunk
- Add `foldChunks :: (chunk -> r -> r) -> r -> (err -> r) -> Chunks chunk err -> r` would be handy?
- Maybe add a rule to fuse `foldChunks` and `safeDecompress`